packages/eslint-plugin/src/stylex-margin-padding-shorthand.js

```diff
@@ -0,0 +1,240 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow strict
+ */
+
+'use strict';
+
+import type {
+  CallExpression,
+  Property,
+  ObjectExpression,
+  Comment,
+} from 'estree';
+
+import type { SourceCode } from 'eslint/eslint-rule';
+
+import type { Token } from 'eslint/eslint-ast';
+
+import splitValue from './utils/splitShorthands.js';
+
+/*:: import { Rule } from 'eslint'; */
+
+type Stack = null | {
+  upper: Stack,
+  prevNode: $ReadOnly<{ ...Property, ... }> | null,
+  prevName: string | null,
+  prevBlankLine: boolean,
+  numKeys: number,
+};
```

Comment:
Should the object type be wrapped in a `$ReadOnly`? Unless you mutate these keys, please do so.